terraso_backend/apps/soilproj/migrations/0001_initial.py

```diff
@@ -0,0 +1,58 @@
+# Generated by Django 4.1.7 on 2023-03-29 21:19
+
+import uuid
+
+import rules.contrib.models
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    initial = True
+
+    dependencies = []
+
+    operations = [
+        migrations.CreateModel(
+            name="Site",
+            fields=[
+                ("deleted_at", models.DateTimeField(db_index=True, editable=False, null=True)),
+                ("deleted_by_cascade", models.BooleanField(default=False, editable=False)),
```

Comment:
question: what is this meant to track? like if it gets deleted because it's part of a project that was deleted?

Reply:
This field is generated by a configuration setting we have in [django-safedelete](https://django-safedelete.readthedocs.io/en/latest/index.html?highlight=safe_delete_field_name#configuration), a Django plugin we use to support "soft" deleting.

The idea is that when we delete a model, the field `deleted_at` is set to the date it was deleted. The plugin modifies the ORM Manager class so that the SELECT queries ignore any rows with this column NOT NULL.

`deleted_by_cascade` is used as well by the plugin to know if the model should be "undeleted" when a linked model is undeleted. So maybe if a site was deleted because the related project was deleted; then if the project is restored, the site should be restored as well. (I'm not sure if that's actually how project deletion will work, but it's an example of what the field is used for).

Reply:
cool, thank you for the explanation! django scares me